spec/parsers/iso_base_media_file_format/utils_spec.rb

```diff
@@ -0,0 +1,632 @@
+require 'matrix'
+require 'spec_helper'
+
+describe FormatParser::ISOBaseMediaFileFormat::Utils do
+  include FormatParser::ISOBaseMediaFileFormat::Utils
+  Box = FormatParser::ISOBaseMediaFileFormat::Box
```

Comment:
If you're testing `Box`, then I'd `describe` it and use it directly as `described_class` and/or `subject` (if it makes sense), rather than manually making it available under a different module's `describe`. I also wouldn't mix in a tested module into a testing block.

Reply:
Not testing `Box`, testing the methods in the `Utils` module and each of those methods take a `box_tree` as an argument - so I need to use the `Box` class to set up the appropriate tests.

Since we're testing the methods in the module, the only options I found to test them were to:
1. `include` the `Utils` module and test the methods directly, as I've done here.
2. Set up a dummy class that includes the module and use that, like:
   ```ruby
   let(:dummy) { (Class.new { include FormatParser::ISOBaseMediaFileFormat::Utils }).new }
   ```

I opted for 1 over 2 because 2 seemed needlessly complex to me and made the tests a little harder to understand, since every method would have to be called as `dummy.method(box_tree)` instead of just calling the method under test directly.

Reply:
If you're testing methods inside a module, you can do that directly, without any intermediary, and provide `Box` as an argument, defined via an overrideable `let`:

```
describe FormatParser::ISOBaseMediaFileFormat::Utils do
  let(:box) { Box.new(...) }

  describe '.method' do
    let(:box) { Box.new(...) }
  end
end
```

I would go for your second approach if I had to test a class/module which had the utils **included**, for whatever reason.

Reply:
I think we're getting wires crossed here. Line 6 here:
```ruby
Box = FormatParser::ISOBaseMediaFileFormat::Box
```
is purely to avoid having to fully qualify the namespace when creating each of the 200+ `Box` instances required to set up the appropriate `box_tree` arguments (which I achieve using `let`) throughout, which greatly improves readability imo.

For example, without that line, this:
```ruby
let(:box_tree) do
  [
    Box.new('moov', 0, 0, nil, [
      Box.new('trak', 0, 0),
      Box.new('trak', 0, 0, nil, [
        Box.new('hdlr', 0, 0, { handler_type: 'soun' }),
      ]),
      Box.new('trak', 0, 0, nil, [
        Box.new('hdlr', 0, 0, { component_type: 'mhlr', component_subtype: 'soun' }),
      ])
    ])
  ]
end
```

instead of:

```ruby
let(:box_tree) do
  [
    FormatParser::ISOBaseMediaFileFormat::Box.new('moov', 0, 0, nil, [
      FormatParser::ISOBaseMediaFileFormat::Box.new('trak', 0, 0),
      FormatParser::ISOBaseMediaFileFormat::Box.new('trak', 0, 0, nil, [
        FormatParser::ISOBaseMediaFileFormat::Box.new('hdlr', 0, 0, { handler_type: 'soun' }),
      ]),
      FormatParser::ISOBaseMediaFileFormat::Box.new('trak', 0, 0, nil, [
        FormatParser::ISOBaseMediaFileFormat::Box.new('hdlr', 0, 0, { component_type: 'mhlr', component_subtype: 'soun' }),
      ])
    ])
  ]
end
```

Reply:
Right, fair enough, that makes sense 😊